### DISCUSSION TITLE
Using Marlin builder

### DISCUSSION THREAD

**Discussion by MetalDracon** [moonglow/FlashForge_Marlin#109] [General]
Hi, your build has been a huge help. I have a Dreamer and have been adapting it to my desired setup and have wanted to modify the marlin build to satisfy my needs. Wile looking at the repository I noticed marlin_builder.sh and thought that this would setup the marlin configuration to the specified inputs. I have no experience reading shell files so I might just be wrong, but if I am right how do I properly run it. Anything helps as so far all attempts to use the script have failed.

**Comment by moonglow** [comment]
Hi @MetalDracon  !

Can you provides me some outputs ? You can use VSCode + PlatformIO plugin to build firmware. Script just run `platformio` from console and sets some global defines. Please refer to official Marlin guide [HERE](https://marlinfw.org/docs/basics/install_rearm.html)

**Comment by MetalDracon** [comment]
I was planning on using VScode to edit the code thinking that the marlin builder tool would give me a starting point, I now realize that it doesn't . I was unaware of how to properly change my directory in the terminal. But I was able to build my Marlin instance using a combination of the global variables you created and the few edits that I planed on adding.
